nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -350,30 +350,26 @@ router.post('/offchain-transaction', async (req, res) => {
     The response from on-chain events converts them to saner string values (e.g. uint64 etc).
     Since we do the transfer off-chain, we do the conversation manually here.
    */
-  const { transactionType, fee } = transaction;
+  const { circuitHash, fee } = transaction;
+
   try {
-    switch (Number(transactionType)) {
-      case 1:
-      case 2:
-      case 3:
-      case 4: {
-        /*
+    const stateInstance = await waitForContract(STATE_CONTRACT_NAME);
+    const circuitInfo = await stateInstance.methods.getCircuitInfo(circuitHash).call();
+    if (circuitInfo.isEscrowRequired) {
+      res.sendStatus(400);
+    } else {
+      /*
           When comparing this with getTransactionSubmittedCalldata,
           note we dont need to decompressProof as proofs are only compressed if they go on-chain.
           let's not directly call transactionSubmittedEventHandler, instead, we'll queue it
          */
-        await enqueueEvent(transactionSubmittedEventHandler, 1, {
-          offchain: true,
-          ...transaction,
-          fee: Number(fee),
-        });
-
-        res.sendStatus(200);
-        break;
-      }
-      default:
-        res.sendStatus(400);
-        break;
+      await enqueueEvent(transactionSubmittedEventHandler, 1, {
+        offchain: true,
+        ...transaction,
+        fee: Number(fee),
```

Comment:
This is an out of scope change for this PR - but this should be `BigInt` in the future if we are support > 64bit fees